mindsdb/integrations/handlers/slack_handler/slack_handler.py

```diff
@@ -0,0 +1,396 @@
+
+from mindsdb.integrations.libs.api_handler import APIHandler, APITable, FuncParser
+from mindsdb.integrations.utilities.sql_utils import extract_comparison_conditions
+
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+    RESPONSE_TYPE
+)
+
+class SlackChannelsTable(APITable):
+    def __init__(self, handler):
+        """
+        Checks the connection is active
+        """
+        super().__init__(handler)
+        self.client = WebClient(token=self.handler.connection_args['token'])
+
+    def select(self, query: ast.Select) -> Response:
+        """
+        Retrieves the data from the channel using SlackAPI
+
+        Args:
+            channel_name
+
+        Returns:
+            conversation_history
+        """
+
+        # Get the channels list and ids
+        channels = self.client.conversations_list(types="public_channel,private_channel")['channels']
+        channel_ids = {c['name']: c['id'] for c in channels}
+        print(channel_ids)
+        
+        # Extract comparison conditions from the query
+        conditions = extract_comparison_conditions(query.where)
+        
+        filters = []
+        params = {}
+        order_by_conditions = {}
+        
+        # Build the filters and parameters for the query
+        for op, arg1, arg2 in conditions:
+            if arg1 == 'channel':
+                if arg2 in channel_ids:
+                    params['channel'] = channel_ids[arg2]
+                else:
+                    raise ValueError(f"Channel '{arg2}' not found")
+
+            elif arg1 == 'limit':
+                if op == '=': 
+                    params['limit'] = int(arg2)
+                else:
+                    raise NotImplementedError(f'Unknown op: {op}')
+
+            else:
+                filters.append([op, arg1, arg2])
+
+        if query.limit:
+            params['limit'] = int(query.limit.value)
+
+        if query.order_by and len(query.order_by) > 0:
+            order_by_conditions["columns"] = []
+            order_by_conditions["ascending"] = []
+
+            for an_order in query.order_by:
+                if an_order.field.parts[1] == "messages":
+                    order_by_conditions["columns"].append("messages")
+
+                    if an_order.direction == "ASC":
+                        order_by_conditions["ascending"].append(True)
+                    else:
+                        order_by_conditions["ascending"].append(False)
+                else:
+                    raise ValueError(
+                        f"Order by unknown column {an_order.field.parts[1]}"
+                    )
+
+        # Retrieve the conversation history
+        try:
+            result = self.client.conversations_history(channel=params['channel'])
+            conversation_history = result["messages"]
+        except SlackApiError as e:
+            print("Error creating conversation: {}".format(e))
```

Comment:
Let's change to log

Reply:
Or raise Error

Reply:
Removed print here, and changed it to log.